File: samples/convert_file/convert_file.js
import fsp from 'fs/promises'
import { file } from '@kittycad/lib/import'

async function main() {
    const body = await fsp.readFile('./ORIGINALVOXEL-3.obj', 'base64')
    // LITTERBOX-END-NON-EDITABLE-SECTION

    const fc = await file.create_file_conversion({
        output_format: 'stl',
        src_format: 'obj',
        body,
    })
    if ('error_code' in fc) throw fc
    console.log(`File conversion id: ${fc.id}`)
    console.log(`File conversion status: ${fc.status}`)

    const outputs = Object.entries(fc.outputs)
    if (outputs.length !== 1) throw Error("Expected one output file")

    // LITTERBOX-START-NON-EDITABLE-SECTION
    for (const [_, output] of outputs) {
        const outputFilePath = './output.stl'
        await fsp.writeFile(outputFilePath, output, 'base64')
    }
}

main()

File: samples/convert_file/convert_file.go
package main

import (
	"fmt"
	"os"

	"github.com/kittycad/kittycad.go"
)

func main() {
	// Create a new client with your token parsed from the environment variable:
	// ZOO_API_TOKEN.
	client, err := kittycad.NewClientFromEnv("your apps user agent")
	if err != nil {
		panic(err)
	}

	fileBytes, _ := os.ReadFile("./ORIGINALVOXEL-3.obj")
	// LITTERBOX-END-NON-EDITABLE-SECTION

	fc, err := client.File.CreateConversion(
		kittycad.FileExportFormatStl,
		kittycad.FileImportFormatObj,
		fileBytes)
	if err != nil {
		panic(err)
	}

	fmt.Println("File conversion id: ", fc.ID)
	fmt.Println("File conversion status: ", fc.Status)

	// LITTERBOX-START-NON-EDITABLE-SECTION
	output_file_path := "./output.stl"
	fmt.Println("Saving output to ", output_file_path)

	output, err := os.Create(output_file_path)
	if err != nil {
		panic(err)
	}
	defer output.Close()

	if len(fc.Outputs) != 1 {
		panic("Expected 1 output file")
	}

	for _, contents := range fc.Outputs {
		if _, err := output.Write(contents.Inner); err != nil {
			panic(err)
		}
	}

	if err := output.Sync(); err != nil {
		panic(err)
	}
}
